bundle/src/main/java/com/adobe/acs/tools/explain_query/impl/ExplainQueryServlet.java

```diff
@@ -202,30 +228,16 @@ protected final void doPost(SlingHttpServletRequest request, SlingHttpServletRes
         } catch (JSONException e) {
             log.error(e.getMessage());
             response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+        } finally {
+            if (logCollectorRegistered) {
+                unregisterLogCollector();
+            }
         }
     }
 
-    @Activate
-    private void activate(Map<String, ?> config, BundleContext context){
-        String[] loggerNames = PropertiesUtil.toStringArray(config.get(PROP_LOGGER_NAMES), null);
-
-        if (loggerNames != null){
-            String pattern = PropertiesUtil.toString(config.get(PROP_MSG_PATTERN), DEFAULT_PATTERN);
-            int msgCountLimit = PropertiesUtil.toInteger(config.get(PROP_LOG_COUNT_LIMIT), DEFAULT_LIMIT);
-            logCollector = new QueryLogCollector(loggerNames, pattern, msgCountLimit, checkMDCSupport(context));
-            logCollectorRegistration = context.registerService(TurboFilter.class.getName(), logCollector, null);
-        }
-    }
-
-    @Deactivate
-    private void deactivate(){
-        if (logCollectorRegistration != null){
-            logCollectorRegistration.unregister();
-        }
-    }
-
-    private JSONObject explainQuery(final Session session, final String statement,
-                                    final String language) throws RepositoryException, JSONException {
+    private JSONObject explainQuery(final Session session, final QueryLogCollector logCollector,
+                                    final String statement, final String language)
+            throws RepositoryException, JSONException {
         final QueryManager queryManager = session.getWorkspace().getQueryManager();
         final JSONObject json = new JSONObject();
         final String collectorKey = startCollection();
```

Comment:
startCollection generates a unique collectorKey for this thread (so concurrent Explain Query runs don't leak log statements into one another)